aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -329,6 +334,52 @@ module aptos_framework::account {
         rotation_capability
     }
 
+    // Offer the account owner's signer capablity to another account
+    public entry fun offer_signer_capability_ed25519(account: &signer, signer_capability_sig_bytes: vector<u8>, account_public_key_bytes: vector<u8>, recipient_address: address) acquires Account {
+        let addr = signer::address_of(account);
+        assert!(exists_at(addr) && exists_at(recipient_address), error::not_found(EACCOUNT_DOES_NOT_EXIST));
+
+        let pubkey = ed25519::new_unvalidated_public_key_from_bytes(account_public_key_bytes);
+        let signer_capability_sig = ed25519::new_signature_from_bytes(signer_capability_sig_bytes);
+
+        // Get the current authentication key of the account and verify that it matches with `account_public_key_bytes`
+        let account_resource = borrow_global_mut<Account>(addr);
```

Comment:
nit: Why declare these variables all the way up here and not where they're needed, closer to line 355?